### ISSUE TITLE
Difficult to close training task on mobile

### ISSUE THREAD

**Issue by chao-mu** [jackstenglein/chess-dojo-scheduler#771]
![image](https://github.com/user-attachments/assets/451d6c89-e985-4c02-8681-ef42669558b6)

**Comment by jackstenglein** [comment]
I think making the task dialog full screen on modal and adding a dedicated close button to the top would probably solve this. Also open to other ideas if you had something in mind already

**Comment by chao-mu** [comment]
Setting a baseline for my changes:

![image](https://github.com/user-attachments/assets/dba04009-8099-4440-8294-d11c3bc5e1a1)
![image](https://github.com/user-attachments/assets/249f8c46-6e94-4e0f-848a-f1c3944d430a)

**Comment by chao-mu** [comment]
There's also the issue of the vertical scrolling and content overwhelm, which I find to be a usability issue.

**Comment by chao-mu** [comment]
I tried full screen and it didn't look good to me. Cross between too much empty space and paradoxically also cramped. 

I like this WAY better. It also makes the update button more visible/less eye movement to find it. Vision starts at the top left naturally and then the update button is immediately apparent.

![image](https://github.com/user-attachments/assets/7ad8a539-d383-4336-afdc-715f458e6477)

![image](https://github.com/user-attachments/assets/3575cfac-36bc-49fd-a089-fbf19370a069)

I made a modal title component and a close button component to be able to easily drop this in other places. I didn't think it was worth wrapping the entirety of MUI's modal stuff.

I also dropped the new close button in the tutorial, which tweaks the spacing a bit for the better. It's pulled more to the top right than the top of the title, which is intentional, but I can tweak if you dislike it.

![image](https://github.com/user-attachments/assets/7290e7ee-f96a-4bec-97e4-0b8552dd2860)
